### ISSUE TITLE
num_eval_steps

### ISSUE THREAD

**Issue by pretidav** [google-research/electra#29]
It is not clear what is num_eval_steps, and what is the loss printed when evaluating performances of 
the pretrained model. 

Is this just the number of batches that are taken to evaluate? 
The loss is just the loss evaluated on the current batch, and hence the final one is just the average among the selected batches? 

Thanks for the answers!

**Comment by pretidav** [comment]
Moreover, is the input data automatically split by the estimator between train and eval? 
It is not clear if the evaluation is done over the same batches used for the training.

**Comment by clarkkev** [comment]
Yes, num_eval_steps is the number of batches that are used to evaluate the model when do_eval is true and the reported metrics are averaged over the batches. Evaluation is just done over the training data. The datasets are large enough that the models don't overfit much so the train loss is a pretty good measure of the model's quality.
